Automattic/wp-calypso
# FSE: Render post title before post content on the editor

#### Changes proposed in this Pull Request

Currently, when editing a full site page, the post title is shown above the header which is not in line with what it will be published.

Since Gutenberg doesn't provide any way to customize the location of the post title, this PR seeks to solve the problem by hiding the post title rendered by default by the editor and making our post content block to render a new post title just before the inner blocks that will determine the post content.

Given that the post content block can be inserted in any post and not only on full site pages, there is a new store that exposes a `isFullSitePage` selector that determines if the currently edited post is a full site page (by checking that current post is a page that has a template assigned and contains a post content block).

| Before | After |
|---|---|
| <img width="1440" alt="Screen Shot 2019-06-25 at 15 05 33" src="https://user-images.githubusercontent.com/1233880/60073182-b3266700-975a-11e9-8ce7-c610b1258aec.png"> | <img width="1440" alt="Screen Shot 2019-06-25 at 15 03 13" src="https://user-images.githubusercontent.com/1233880/60073061-604caf80-975a-11e9-941f-f533122b8ff5.png"> |

#### Testing instructions

* Apply this branch to your FSE testing environment.
* Trash all templates and template parts.
* Deactivate and then reactivate the FSE plugin.
* Create a new page. It doesn't matter which template you select. Add some content to that page.
* Assign the default _Page Template_ to the page by using the templates button on the toolbar, next to Settings.
<img width="289" alt="Screen Shot 2019-06-25 at 15 09 12" src="https://user-images.githubusercontent.com/1233880/60073368-36e05380-975b-11e9-8676-94f03437b870.png">

* Publish the page and refresh the editor.
  * The reload is needed because at the moment, when you first select a template, the template parts are not applied live. The same when you change a template. That's still a work in progress.
* Make sure the title of the page is displayed below the header template part and at the beginning of the post content block.
* Try changing the page title or the permalink and confirm that everything is working as expected.
* Try now inserting a post content blog a not full site page (i.e. in a regular post or in template) and verify the post title is visible in the editor.

Fixes #34072
